packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -503,25 +503,30 @@ final GradleHandledError minCompileSdkVersionHandler = GradleHandledError(
   eventLabel: 'min-compile-sdk-version',
 );
 
+final RegExp _agpJavaError = RegExp(r'Android Gradle plugin requires Java\s+\d+\s+to run');
```

Comment:
FWIW the android code assumes that this will be some number of characters not a number. 
https://cs.android.com/android-studio/platform/tools/adt/idea/+/mirror-goog-studio-main:project-system-gradle/src/com/android/tools/idea/gradle/project/sync/errors/UnsupportedJavaVersionForAgpIssueChecker.kt?q=%22Android%20Gradle%20plugin%20requires%20Java%22 

Given that I think we should make our regex accept any chars instead of a number.

Reply:
done